mobile_config/tests/integrations/common/gateway_metadata_db.rs

```diff
@@ -1,76 +1,114 @@
+                    INSERT INTO mobile_hotspot_infos (
+                        asset, location, device_type, created_at,
+                        refreshed_at, deployment_info, num_location_asserts
+                    )
+                    "#,
+                );
+
+                qb.push_values(chunk, |mut b, gw| {
+                    let num_locations = if gw.location.is_some() {
+                        Some(1)
+                    } else {
+                        Some(0)
+                    };
+
+                    let device_type_json: serde_json::Value =
+                        serde_json::from_str(&gw.device_type).unwrap();
+                    let deployment_info_json: serde_json::Value =
+                        serde_json::from_str(gw.deployment_info.as_deref().unwrap_or("null"))
+                            .unwrap();
+
+                    b.push_bind(&gw.asset)
+                        .push_bind(gw.location)
+                        .push_bind(device_type_json)
+                        .push_bind(gw.created_at)
+                        .push_bind(gw.refreshed_at)
+                        .push_bind(deployment_info_json)
+                        .push_bind(num_locations);
+                });
+
+                qb.build().execute(&mut *tx).await?;
+
+                // insert key_to_assets
+                let mut qb1 = QueryBuilder::<Postgres>::new(
+                    r#"
+                    INSERT INTO key_to_assets (
+                        asset, entity_key
+                    )
+                    "#,
+                );
+
+                qb1.push_values(chunk, |mut b, gw| {
+                    let b58 = bs58::decode(gw.key.to_string()).into_vec().unwrap();
+                    b.push_bind(&gw.asset).push_bind(b58);
+                });
+
+                qb1.build().execute(&mut *tx).await?;
+
+                tx.commit().await?;
+
+                Ok::<_, anyhow::Error>(())
+            }
+        })
+        .await?;
+
+    Ok(())
 }
 
-async fn insert_mobile_hotspot_infos(
+pub async fn update_gateway(
     pool: &PgPool,
     asset: &str,
-    location: Option<i64>,
-    device_type: &str,
-    created_at: DateTime<Utc>,
-    refreshed_at: Option<DateTime<Utc>>,
-    deployment_info: Option<&str>,
-) {
-    let num_locations = if location.is_some() { Some(1) } else { Some(0) };
+    location: i64,
+    refreshed_at: DateTime<Utc>,
+) -> anyhow::Result<()> {
     sqlx::query(
         r#"
-            INSERT INTO
-"mobile_hotspot_infos" ("asset", "location", "device_type", "created_at", "refreshed_at", "deployment_info", "num_location_asserts")
-            VALUES
-($1, $2, $3::jsonb, $4, $5, $6::jsonb, $7);
-    "#,
+        UPDATE mobile_hotspot_infos
+        SET location = $1,
+            num_location_asserts = $2,
+            refreshed_at = $3
+        WHERE asset = $4
+        "#,
     )
-    .bind(asset)
     .bind(location)
-    .bind(device_type)
-    .bind(created_at)
+    .bind(2)
```

Comment:
Can we pass this in as an argument? If we want to test for a gateway that has already had 2 location asserts, or we use this function having expected more, it will be very confusing to find.